patches/server/0752-Fix-and-optimize-legacy-world-conversion.patch

```diff
@@ -0,0 +1,163 @@
+         }
+ 
++        // Paper start - copied from vanilla before below CB diff
++        public File getDimensionPathForLegacyConversion(ResourceKey<Level> key) {
++            return DimensionType.getStorageFolder(key, this.levelPath.toFile());
++        }
++        // Paper end
++
+         public File getDimensionPath(ResourceKey<Level> key) {
+             return LevelStorageSource.getFolder(this.levelPath.toFile(), this.dimensionType); // CraftBukkit
+         }
+diff --git a/src/main/java/net/minecraft/world/level/storage/McRegionUpgrader.java b/src/main/java/net/minecraft/world/level/storage/McRegionUpgrader.java
+index 87705fc8ee32016bf5ca533eb278bf32df08d3a3..2b7ac4d0f0695302f2fa8b04c0511f2aadf91675 100644
+--- a/src/main/java/net/minecraft/world/level/storage/McRegionUpgrader.java
++++ b/src/main/java/net/minecraft/world/level/storage/McRegionUpgrader.java
+@@ -35,13 +35,29 @@ public class McRegionUpgrader {
+     private static final String MCREGION_EXTENSION = ".mcr";
+ 
+     static boolean convertLevel(LevelStorageSource.LevelStorageAccess storageSession, ProgressListener progressListener) {
++        // Paper start
++        progressListener = new ProgressListener() {
++            @Override
++            public void progressStartNoAbort(net.minecraft.network.chat.Component title) {}
++            @Override
++            public void progressStart(net.minecraft.network.chat.Component title) {}
++            @Override
++            public void progressStage(net.minecraft.network.chat.Component task) {}
++            @Override
++            public void progressStagePercentage(int percentage) {}
++            @Override
++            public void stop() {}
++        };
++        // Paper end
+         progressListener.progressStagePercentage(0);
+         List<File> list = Lists.newArrayList();
+         List<File> list2 = Lists.newArrayList();
+         List<File> list3 = Lists.newArrayList();
+-        File file = storageSession.getDimensionPath(Level.OVERWORLD);
+-        File file2 = storageSession.getDimensionPath(Level.NETHER);
+-        File file3 = storageSession.getDimensionPath(Level.END);
++        // Paper start
++        File file = storageSession.getDimensionPathForLegacyConversion(Level.OVERWORLD);
++        File file2 = storageSession.getDimensionPathForLegacyConversion(Level.NETHER);
++        File file3 = storageSession.getDimensionPathForLegacyConversion(Level.END);
++        // Paper end
+         LOGGER.info("Scanning folders...");
+         addRegionFiles(file, list);
+         if (file2.exists()) {
+@@ -88,15 +104,65 @@ public class McRegionUpgrader {
+     }
+ 
+     private static void convertRegions(RegistryAccess.RegistryHolder registryManager, File directory, Iterable<File> files, BiomeSource biomeSource, int i, int j, ProgressListener progressListener) {
++        // Paper start
++        convertRegionsThreaded(registryManager, directory, files, biomeSource, i, j, progressListener);
++        /*
```

Comment:
Afaik (according to [MM's comment](https://github.com/PaperMC/Paper/pull/6450#discussion_r693262113)) there is a move away from comment blocks for, at least in my opinion, a pretty valid reason.
Should this also apply here ?

Reply:
Yeah, I know I’m a big min diff guy, but we’ve been burned by changes in block comments applying without a conflict before, and so I now encourage line comments.